napalm_eos/eos.py

```
@@ -985,16 +985,20 @@ def get_mac_address_table(self):
     def get_route_to(self, destination='', protocol=''):
         routes = dict()
 
+        if protocol.lower() not in ['', 'bgp', 'connected', 'isis', 'ospf', 'rip', 'static']:
```

Let's remove this.

---

More than happy to, was just trying to match up closer to the junos reference spec.  Do we want to try to throw a special UnknownProtocol exception or let the command parser raise an exception when an unknown protocol is passed in, instead?

---

That would be the most optimal, but for the moment let's just raise ```TypeError```.
And do check only if the protocol is specified, otherwise will return all possible route.s

---

Can you add in that list also ```direct``` and when user requests ```direct``` to rewrite the value of the ```protocol``` var to ```connected```?

---

Eeerm... I know I'll contradict ourselves (did we discuss that before?) but I think we should remove this line completely. I don't see any reason why we'd limit to some certain protocols only. I will make the necessary adjustment on the other drivers in order to avoid this limitation!

---

My approach was to enumerate ALL protocols supported by the driver, and raise TypeError for a bad one, instead of having the underlying driver pyeapi raise a CommandError.  Either way works for me